clap_derive/src/derives/arg_enum.rs

```diff
@@ -91,23 +92,32 @@ fn lits(
             }
         })
         .flat_map(|(variant, attrs)| {
-            let mut ret = vec![(attrs.cased_name(), variant.ident.clone())];
+            let mut ret = vec![(attrs.cased_name(), variant.ident.clone(), attrs.clone())];
 
             attrs
                 .enum_aliases()
                 .into_iter()
-                .for_each(|x| ret.push((x, variant.ident.clone())));
+                .for_each(|x| ret.push((x, variant.ident.clone(), attrs.clone())));
 
             ret
         })
         .collect::<Vec<_>>()
 }
 
-fn gen_variants(lits: &[(TokenStream, Ident)]) -> TokenStream {
-    let lit = lits.iter().map(|l| &l.0).collect::<Vec<_>>();
+fn gen_variants(lits: &[(TokenStream, Ident, Attrs)]) -> TokenStream {
+    let lit = lits
+        .iter()
+        .map(|(name, _, attrs)| {
+            let fields = attrs.field_methods_filtered(&["alias"]);
+            quote! {
+                clap::ArgValue::new(#name)
+                #fields
+            }
```

Comment:
Now that we have an `ArgValue`, should we move support for `alias` from the derive to `ArgValue`?  This will make clap more self-consistent and simplify parts of the derive (no special casing for `alias` with `field_methods_filtered`).

On the other hand, its relatively trivial for a user to implement aliases like the derive does, reducing the value gained from adding it.

Reply:
we could, but then we should add some code to still easily get an iterator with all "possibel_values" in other places so we don't have to "expand" ArgValue every time.

On the other hand this might even pose an advantage if we want to indicate in any way that multiple values share their meaning in help messages etc.

But does it pose any other advantage to people using the builder? For derive this makes parsing easier, but it doesn't change anything (except only needing to define `hidden` and `about` once) when you parse arguments using the builder? If I understand this correctly.

Reply:
The advantage for the builder API is consistency; they are used to `alias()` and they get it in each place.  Like I said, its fairly minor.

Reply:
What would be a good way for implementing? because it would mean more code in each place possible_values are used. we could maybe implement something to get from `Iter<ArgValue>` to `Iter<&str>` but that would only solve the case where we don't need about, (hide could be implemented so that this would return only the visible values).

we could also use `Flatten` but then we would either need a `FlatArgValue` without aliases that `ArgValue` can be `IntoIterator` for, or we would have `impl IntoIterrator<Item=ArgValue> for ArgValue` but that would probably not really yield something consistent, as there is no indication that the resulting iterrator is flattened.

I think this would improve the builder api, but not sure what would be the best way about implementing this internally.

Reply:
As one example, I suspect we could change
```rust
            Some(format!(
                "(({}))",
                values
                    .iter()
                    .filter_map(|value| {
                        if value.is_hidden() {
                            None
                        } else {
                            Some(format!(
                                r#"{name}\:"{about}""#,
                                name = escape_value(value.get_name()),
                                about = value.get_about().map(escape_help).unwrap_or_default()
                            ))
                        }
                    })
                    .collect::<Vec<_>>()
                    .join("\n")
            ))
```
to
```rust
            Some(format!(
                "(({}))",
                values
                    .iter()
                    .filter(|value| !value.is_hidden())
                    .flat_map(|value| {
                        [format_value(value.get_name(), value.get_about()].into_iter().chain(
                            value.get_aliases().map(|alias| format_value(alias), value.get_about()))
                        }
                    })
                    .collect::<Vec<_>>()
                    .join("\n")
            ))

fn format_value(name: &str, about: Option<&str:>) {
                            format!(
                                r#"{name}\:"{about}""#,
                                name = escape_value(alias),
                                about = about.map(escape_help).unwrap_or_default()
                            )
}
```